src/NuGet.Core/NuGet.Build.Tasks/GenerateTemporaryTargetAssembly2.cs

```diff
@@ -0,0 +1,523 @@
+            set { _compileTypeName = value; }
+        }
+
+
+        /// <summary>
+        /// ReferencePath
+        ///    A list of resolved reference assemblies.
+        ///    The list will replace the original Reference item list in generated project file.
+        /// </summary>
+        public ITaskItem[] ReferencePath
+        {
+            get { return _referencePath; }
+            set { _referencePath = value; }
+        }
+
+        /// <summary>
+        /// ReferencePathTypeName
+        ///   The appropriate item name which is used to keep the Reference list in managed compiler task.
+        ///   It is "ReferencePath" for now.
+        ///   
+        ///   Adding this property is to make the type name configurable, if it is changed, 
+        ///   No code is required to change in this task, but set a new type name in project file.
+        /// </summary>
+        [Required]
+        public string ReferencePathTypeName
+        {
+            get { return _referencePathTypeName; }
+            set { _referencePathTypeName = value; }
+        }
+
+
+        /// <summary>
+        /// IntermediateOutputPath
+        /// 
+        /// The value which is set to IntermediateOutputPath property in current project file.
+        /// 
+        /// Passing this value explicitly is to make sure to generate temporary target assembly 
+        /// in expected directory.  
+        /// </summary>
+        [Required]
+        public string IntermediateOutputPath
+        {
+            get { return _intermediateOutputPath; }
+            set { _intermediateOutputPath = value; }
+        }
+
+        /// <summary>
+        /// AssemblyName
+        /// 
+        /// The value which is set to AssemblyName property in current project file.
+        /// Passing this value explicitly is to make sure to generate the expected 
+        /// temporary target assembly.
+        /// 
+        /// </summary>
+        [Required]
+        public string AssemblyName
+        {
+            get { return _assemblyName; }
+            set { _assemblyName = value; }
+        }
+
+        /// <summary>
+        /// CompileTargetName
+        /// 
+        /// The msbuild target name which is used to generate assembly from source code files.
+        /// Usually it is "CoreCompile"
+        /// 
+        /// </summary>
+        [Required]
+        public string CompileTargetName
+        {
+            get { return _compileTargetName; }
+            set { _compileTargetName = value; }
+        }
+
+        #endregion Public Properties
+
+        //------------------------------------------------------
+        //
+        //  Private Methods
+        //
+        //------------------------------------------------------
+
+        #region Private Methods
+
+        //
+        // Remove specific items from project file. Every item should be under an ItemGroup.
+        //
+        private void RemoveItemsByName(XmlDocument xmlProjectDoc, string sItemName)
```

Comment:
There is a lot I'd like to critique about this method, but I'll forbear since I suspect you just copied it and don't want to change it.

Reply:
yes, copied this method.